contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -427,152 +750,119 @@ NXopt(NXobject):
           layer (the one on the front surface, on which the light incident).
           The order must be consistent with layer_structure
       atom_types:
+        exists: optional
         doc: |
           List of comma-separated elements from the periodic table that are
           contained in the sample. If the sample substance has multiple
           components, all elements from each component must be included in
           'atom_types'.
-      sample_history:
-        doc: |
-          Ideally, a reference to the location or a unique (globally
-          persistent) identifier (e.g.) of e.g. another file which gives
-          as many as possible details of the material, its microstructure,
-          and its thermo-chemo-mechanical processing/preparation history.
-          In the case that such a detailed history of the sample is not
-          available, use this field as a free-text description to specify
-          details of the sample and its preparation.
       preparation_date(NX_DATE_TIME):
         exists: recommended
         doc: |
-          ISO8601 date with time zone (UTC offset) specified.
-      substrate:
+          ISO 8601 time code with local time zone offset to UTC information
+          when the specimen was prepared.
+          
+          Ideally, report the end of the preparation, i.e. the last known timestamp when
+          the measured specimen surface was actively prepared.
+      history(NXhistory):
         exists: recommended
         doc: |
-          Description of the substrate.
-      sample_orientation:
-        exists: optional
-        doc: |
-          Specify the sample orientation.
-    data_collection(NXprocess):
-      doc: |
-        Measured data, data errors, and varied parameters. If reference data
-        were measured they should be considered a separate experiment and a
-        link to its NeXus file should be added in reference_data_link.
-      data_identifier(NX_NUMBER):
-        doc: |
-          An identifier to correlate data to the experimental conditions,
-          if several were used in this measurement; typically an index of 0-N.
-      data_type:
-        doc: |
-          Select which type of data was recorded, for example intensity,
-          reflectivity, transmittance, Psi and Delta etc.
-          It is possible to have multiple selections. The enumeration list
-          depends on the type of experiment and may differ for different
-          application definitions.
-        enumeration: [intensity, reflectivity, transmittance, Psi/Delta, tan(Psi)/cos(Delta), Mueller matrix, Jones matrix, N/C/S, raw data]
-      
-      # This should be a required field, but is set to 'optional' for the moment
-      NAME_spectrum(NX_FLOAT):
-        exists: optional
-        unit: NX_ANY
+          A set of activities that occurred to the sample prior to/during the experiment.
+        history_id(NXidentifier):
```

Comment:
This is a good idea that should be kept for the generic `NXhistory` (if we decide to extend it in the future).

Reply:
So, it may should be included in NXhistory? Just to clarify, as I maybe dont understand the impact of this

Reply:
Yeah, please make a PR for this (or include this in this PR, whatever works better).